smartapps/smartthings/ecobee-connect.src/ecobee-connect.groovy

```diff
@@ -463,29 +463,33 @@ def pollChildren(child = null) {
 }
 
 // Poll Child is invoked from the Child Device itself as part of the Poll Capability
-def pollChild(child){
-
-	if (pollChildren(child)){
-		if (!child.device.deviceNetworkId.startsWith("ecobee_sensor")){
-			if(atomicState.thermostats[child.device.deviceNetworkId] != null) {
-				def tData = atomicState.thermostats[child.device.deviceNetworkId]
-				log.info "pollChild(child)>> data for ${child.device.deviceNetworkId} : ${tData.data}"
-				child.generateEvent(tData.data) //parse received message from parent
-			} else if(atomicState.thermostats[child.device.deviceNetworkId] == null) {
-				log.error "ERROR: Device connection removed? no data for ${child.device.deviceNetworkId}"
-				return null
+def pollChild(){
+
+	def devices = getChildDevices()
+
+	if (pollChildren()){
+		devices.each { child ->
+			log.info "***found $child"
+			if (!child.device.deviceNetworkId.startsWith("ecobee_sensor")){
```

Comment:
So, as I understand.  You just iterate over all the devices looking for ecobees, instead of accepting a child device and generating the event directly?